### PR TITLE
Firewall services dialog ports

### PR DESCRIPTION
**Pull request by Gundersanne**
image

### CODE REVIEW

**Comment by martinpitt** [comment]
Technically and conceptually this looks excellent, thank you! The design might still need some love though?

**Comment by Gundersanne** [comment]
alternative:
image

**Comment by garrett** [comment]
Ouch.

But yeah, I agree that showing the ports makes sense.

We probably want to make it look similar to the list @ https://github.com/cockpit-project/cockpit/pull/11407#issuecomment-474015347, with the title as the first line and the details as a secondary line.

Here's a screenshot from the linked comment:

54549149-c2528780-49a9-11e9-98b1-213bfea7b751

The downside of placing the port info below the title is that it'd make things taller. However, as the TCP and UDP are currently stacked as two lines your screenshots above, it's already 2 lines tall... so it isn't really worse.

**Comment by Gundersanne** [comment]
@garrett 

opinions? I tihnk the example you show is with our `Listing`.

image

**Comment by garrett** [comment]
@Gundersanne: Wow, that looks a _lot_ better! I like it.

I guess this is using a PF list instead of a custom list?

**Comment by garrett** [comment]
@Gundersanne: I made some changes in https://github.com/cockpit-project/cockpit/pull/11420 (custom ports in services dialog) to the services list (porting to ct-form-layout among other things)... and now the PRs conflict with each other. Hopefully it shouldn't be difficult to merge the changes in to whichever PR lands second.

**Comment by Gundersanne** [comment]
@garret This is the PF list but with 2 css rules overwriting default styles, one removing padding and one reducing padding.

File: pkg/networkmanager/firewall.jsx [hunk 1]
```diff
@@ -120,6 +121,23 @@ class SearchInput extends React.Component {
...
+        if (port.protocol === "tcp")
+            tcpPorts.push(port.port);
+        else
+            udpPorts.push(port.port);
+    }
+    return (
+        <React.Fragment>
+            { tcpPorts.length > 0 && <Row>&nbsp;<strong>TCP: </strong>{ tcpPorts.join(', ') }</Row> }
+            { udpPorts.length > 0 && <Row>&nbsp;<strong>UDP: </strong>{ udpPorts.join(', ') }</Row> }
+        </React.Fragment>
```
Comment (martinpitt) [inline pkg/networkmanager/firewall.jsx]: Can you please add some check for the expected ports to the tests? The ones we add in the tests should have predictable ports across all OSes.

Comment (Gundersanne) [reply pkg/networkmanager/firewall.jsx]: https://github.com/cockpit-project/cockpit/blob/43b58346eaa8f9d3f8bd41b9703f455a472a1eca/test/verify/check-networking-firewall#L148 added!

**Comment by Gundersanne** [comment]
@pitti Somewhere the horizontal scrollbar sneaked in, I might make a follow up to fix that design. Just letting the `<small>` field use `overflow: hidden`  might be best.

**Pull request merged**